### PR TITLE
#22 get plans and display in plan gallery

### PR DESCRIPTION
**Pull request by wehnerts**
Get plans implemented in backend and frontend. Including tests.  some minor changes in css.

Allen einen schönen Abend :-)

### CODE REVIEW

File: backend/src/main/java/de/wehnerts/backend/dto/PlanItemDto.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+
+
+
```
Comment (attilaDora) [inline backend/src/main/java/de/wehnerts/backend/dto/PlanItemDto.java:12]: delete lines

File: backend/src/main/java/de/wehnerts/backend/service/PlanItemService.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import de.wehnerts.backend.model.PlanItem;
+import de.wehnerts.backend.repository.ActionItemRepo;
+import de.wehnerts.backend.repository.PlanItemRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
```
Comment (attilaDora) [inline backend/src/main/java/de/wehnerts/backend/service/PlanItemService.java:13]: delete line

File: backend/src/test/java/de/wehnerts/backend/service/PlanItemServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                .price ("ne Mark")
+                .homepage("www.de")
+                .build();
+        when(planItemRepo.findAll()).thenReturn(List.of(plan1, plan2));
+
+
+        when(actionItemRepo.findById("1234567")).thenReturn(Optional.of(item1));
+        when(actionItemRepo.findById("8834567")).thenReturn(Optional.of(item2));
+
+
```
Comment (attilaDora) [inline backend/src/test/java/de/wehnerts/backend/service/PlanItemServiceTest.java:77]: delete line

File: frontend/src/components/PlanItem/PlanItemCard.css [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    margin: 10px 10px 10px 10px;
+    box-shadow: #847d61 4px 4px ;
+    align-items: center ;
+    flex-direction: column;
+    border-color: #f4e07f;
+    border-width: medium;
+    border-style: solid;
+
+    border-bottom-left-radius: 5px 5px;
+    border-top-right-radius: 5px 5px;
```
Comment (attilaDora) [inline frontend/src/components/PlanItem/PlanItemCard.css:14]:
```suggestion
    border-radius: 5px;
    
```

File: frontend/src/components/PlanItem/PlanItemCard.css [hunk 5]
```diff
@@ -0,0 +1,21 @@
+.plan-item-card{
+    display: flex;
+    background-color: #F27059;
+    color:#2C1A1D;
+    margin: 10px 10px 10px 10px;
```
Comment (attilaDora) [inline frontend/src/components/PlanItem/PlanItemCard.css:5]:
```suggestion
    margin: 10px;
```

File: frontend/src/components/ActionItem/ActionItemCard.css [hunk 6]
```diff
@@ -1,6 +1,6 @@
 .action-item-card{
     display: flex;
-    background-color: #0EB1D2;
+    background-color: #F27059;
     color:#2C1A1D;
     margin: 10px 10px 10px 10px;
```
Comment (attilaDora) [inline frontend/src/components/ActionItem/ActionItemCard.css:5]:
```suggestion
    margin: 10px;
```

File: frontend/src/components/ActionItem/EditActionItem.tsx [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import React, {FormEvent,  useState} from "react";
 import {toast} from "react-toastify";
 import './NewActionItem.css'
-import {ActionItem} from "../model/ActionItem";
+import {ActionItem} from "../../model/ActionItem";
```
Comment (attilaDora) [inline frontend/src/components/ActionItem/EditActionItem.tsx:6]: delete line

File: frontend/src/hooks/usePlanItems.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+import {toast} from "react-toastify";
+import {getAllPlanItems} from "../services/api-service-plan";
+
+export default function usePlanItems(){
+    const [planItems, setPlanItems]= useState<PlanItem[]>([])
+
+    useEffect(()=>{
+        getAllPlanItems()
+            .then(allPlanI => setPlanItems(allPlanI))
+            .catch(()=>toast.error("Connection failed! Please retry later"))
```
Comment (attilaDora) [inline frontend/src/hooks/usePlanItems.ts:12]:
```suggestion
            .catch(()=>toast.error("Connection failed! Please try again later"))
```

**Review by attilaDora — APPROVED** [review]
I found some blank lines and added some small updates that I think make it nicer.

**Review by wehnerts — COMMENTED** [review]
Danke für den Review, die Margin mit 4  werten sind später zur genaueren Ausrichtung erforderlich. Die leerzeilen kontrolliere ich noch mal.

**Pull request merged**